Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/MainController.java

```diff
@@ -128,14 +146,104 @@
     public PortalSession portalSession;
     private Map<String, PortalPlugin> portalPluginMap;
 
+    static {
+        String encPassphStr = ApromoreEnvUtils.getEnvPropValue(
+                SYMMETRIC_KEY_SECRET_ENV_KEY, "encryption passphrase could *not* be attained from environment");
+
+        MainController.encKey = encPassphStr;
+        LOGGER.info("\n\nInitialised main controller and obtained encKey from environment\n");
+
+        // For safety/clear
+        encPassphStr = null;
+    }
+
+    public static MainController getController() {
+        return controller;
+    }
+
     public MainController() {
-        qe = EventQueues.lookup(Constants.EVENT_QUEUE_REFRESH_SCREEN, EventQueues.SESSION, true);
+        LOGGER.info("\n\n>>> In MainController() default constructor");
+
+        final boolean usingKeycloak = config.isUseKeycloakSso();
+        LOGGER.info("\n\nUsing keycloak: {}", usingKeycloak);
+
+        if (usingKeycloak) {
+            final Object nativeRequest = Executions.getCurrent().getNativeRequest();
+            LOGGER.info("\n### nativeRequest {}", nativeRequest);
+            final SecurityContextHolderAwareRequestWrapper servReqWrapper =
+                    (SecurityContextHolderAwareRequestWrapper) nativeRequest;
+            final String appAuthHeader = servReqWrapper.getHeader("App_Auth");
+            final String signedAppAuthHeader = servReqWrapper.getHeader("Signed_App_Auth");
+            LOGGER.info("\n### appAuthHeader {}", appAuthHeader);
+            LOGGER.info("\n### signedAppAuthHeader {}", signedAppAuthHeader);
+        }
```

Comment:
Refactor this into the security service.

Reply:
This has been refactored into security helper [DONE].